velox/common/encode/EncoderBase.h

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <exception>
+#include <map>
+#include <string>
+
+#include <folly/Range.h>
+
+namespace facebook::velox::encoding {
+
+class EncoderException : public std::exception {
+ public:
+  explicit EncoderException(const char* msg) : msg_(msg) {}
+  const char* what() const noexcept override {
+    return msg_;
+  }
+
+ protected:
+  const char* msg_;
+};
+
+/// EncoderBase class for all binary encoding scheme for reversibly translating
+/// between byte sequences and printable ASCII strings specified by RFC 4648.
+class EncoderBase {
```

Comment:
The more I look at this, I realized that all functions are static and EncoderBase has no common state shared between Base32 and Base64. It has common utilities. I would rather remove the class and just rename files and EncoderUtils.
Make all definitions global within the namespace `facebook::velox::encoding`.
Please lmk what you think.

Reply:
The idea is that the respective encoder should inherit from the EncoderBase. Therefore, we created a base class that could have all the common functionality and leave the specialisation to the derived classes. For example, calculating the padding depends on encoding specific sizes that the base function can calculate (based on member values set in the specialised classes). I think it's best to go with the base class concept rather than the utility class; this was discussed during the internal review, and we came up with the above decision. please lmk what you think.

Reply:
I understand your intention but having class and making all methods `public static` isn't any different than having a global method. For instance, `baseReverseLookup(kBase, src[0], reverse_lookup)`, this function call takes the base as a parameter though it is called within the context of the class. I don't see a use case for a base class here.
It will reduce the code lines if we eliminate that. Make it `EncoderCommonUtils`. @aditi-pandit lmk what you think.

Reply:
@karteekmurthys Updated the code, let me know what you think.